omod/src/test/java/org/openmrs/web/controller/ForgotPasswordFormControllerTest.java

```diff
@@ -0,0 +1,268 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.controller;
+
+
+import org.openmrs.web.test.BaseModuleWebContextSensitiveTest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.openmrs.api.context.Context;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+
+
+/**
+ * Test the different aspects of
+ * {@link org.openmrs.web.controller.ForgotPasswordFormController}
+ */
+
+public class ForgotPasswordFormControllerTest extends BaseModuleWebContextSensitiveTest {
+	
+	protected static final String TEST_DATA = "org/openmrs/web/controller/include/ForgotPasswordFormControllerTest.xml";
+	
+	@Before
+	public void runBeforeEachTest() throws Exception {
+		executeDataSet(TEST_DATA);
+		Context.logout();
+	}
+	
+	/**
+	 * Check to see if the admin's secret question comes back
+	 *
+	 * @throws Exception
+	 */
+	
+	@Test
+	public void shouldNotNotFailOnNotFoundUsername() throws Exception {
+		
+		ForgotPasswordFormController controller = new ForgotPasswordFormController();
+		
+		
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		MockHttpServletResponse response = new MockHttpServletResponse();
+		
+		request.setParameter("uname", "validuser");
+		request.setMethod("POST");
+		
+		controller.handleRequest(request, response);
+		
+		Assert.assertEquals("validuser", request.getAttribute("uname"));
+		Assert.assertEquals("valid secret question", request.getAttribute("secretQuestion"));
+	}
+	
+	
+	@Test
+	public void shouldAuthenticateAsUserWithValidSecretQuestion() throws Exception {
```

Comment:
Nope the system in not changing it. This password is assigned to log him in. Basically the 'Old password' section is filled up. He has to enter the new password. That's it